src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatProvider.cs

```diff
@@ -0,0 +1,328 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Threading;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    /// <summary>
+    /// Implementation of health heartbeat functionality.
+    /// </summary>
+    internal class HeartbeatProvider : IDisposable, IHeartbeatProvider
+    {
+        /// <summary>
+        /// The default interval between heartbeats if not specified by the user. Left public for use in unit tests.
+        /// </summary>
+        public static int DefaultHeartbeatIntervalMs = 5000;
```

Comment:
why not `TimeSpan`?

Reply:
Agreed, this should be a TimeSpan. _Done._